tests/components/homeassistant_hardware/test_util.py

```diff
@@ -280,3 +286,99 @@ async def test_get_otbr_addon_firmware_info_failure_bad_options(
     )
 
     assert (await get_otbr_addon_firmware_info(hass, otbr_addon_manager)) is None
+
+
+@pytest.mark.parametrize(
+    ("app_type", "firmware_version", "expected_fw_info"),
+    [
+        (
+            FlasherApplicationType.EZSP,
+            "1.0.0",
+            FirmwareInfo(
+                device="/dev/ttyUSB0",
+                firmware_type=ApplicationType.EZSP,
+                firmware_version="1.0.0",
+                source="probe",
+                owners=[],
+            ),
+        ),
+        (
+            FlasherApplicationType.EZSP,
+            None,
+            FirmwareInfo(
+                device="/dev/ttyUSB0",
+                firmware_type=ApplicationType.EZSP,
+                firmware_version=None,
+                source="probe",
+                owners=[],
+            ),
+        ),
+        (
+            FlasherApplicationType.SPINEL,
+            "2.0.0",
+            FirmwareInfo(
+                device="/dev/ttyUSB0",
+                firmware_type=ApplicationType.SPINEL,
+                firmware_version="2.0.0",
+                source="probe",
+                owners=[],
+            ),
+        ),
+        (None, None, None),
+    ],
+)
+async def test_probe_silabs_firmware_info(
+    app_type: FlasherApplicationType | None,
+    firmware_version: str | None,
+    expected_fw_info: FirmwareInfo | None,
+) -> None:
+    """Test getting the firmware info."""
+
+    def probe_app_type() -> None:
+        mock_flasher.app_type = app_type
+
+        if firmware_version is not None:
+            mock_flasher.app_version = FlasherVersion(firmware_version)
+        else:
+            mock_flasher.app_version = None
```

Comment:
if you make the `firmware_version` of type `FlasherVersion | None`, you can remove this if statement

Reply:
Thanks! I've simplified it in https://github.com/home-assistant/core/pull/138874/commits/6a7cef15e65811e07cb467079a099f6e36feae7e.